api/src/main/java/com/datastrato/gravitino/authorization/Resources.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import java.util.Objects;
+
+/** The helper class for {@link Resource}. */
+public class Resources {
+
+  /**
+   * Create the {@link Resource} with the given names.
+   *
+   * @param names The names of the resource
+   * @return The created {@link Resource}
+   */
+  public static Resource of(String... names) {
+    if (names == null) {
+      throw new IllegalArgumentException("Cannot create a resource with null names");
+    }
+
+    if (names.length == 0) {
+      throw new IllegalArgumentException("Cannot create a resource with no names");
+    }
+
+    Resource parent = null;
+    for (String name : names) {
+      if (name == null) {
+        throw new IllegalArgumentException("Cannot create a resource with null name");
+      }
+
+      if (name.equals("*")) {
+        throw new IllegalArgumentException(
+            "Cannot create a resource with `*` name. If you want to use a resource which represents all catalogs,"
+                + " you use the method `ofAllCatalogs`."
+                + " If you want to create an another resource which represents all entities,"
+                + " you can use its parent entity, For example,"
+                + " if you want to have read table privileges of all tables of `catalog1.schema1`,"
+                + " you can use add `read table` privilege for `catalog1.schema1` directly");
```

Comment:
Frankly saying, It's weird to add table privilege to the schema, it will confuse the users.

Reply:
Now, we give a resource a privileges list. It's a little weird add `load catalog` and `load schema` privilege to `catalog1.*`.

Reply:
Databricks follows similar style. Parent doesn't use *. https://docs.databricks.com/en/sql/language-manual/sql-ref-privileges.html